kitsune/wiki/utils.py

```diff
@@ -183,3 +190,21 @@ def get_visible_document_or_404(user, **kwargs):
 
 def get_visible_revision_or_404(user, **kwargs):
     return get_object_or_404(Revision.objects.visible(user, **kwargs))
+
+
+def get_visible_document_or_fallback_or_404(user, locale, slug):
+    """Return a document with its fallback code, or 404."""
+    try:
+        return (Document.objects.get_visible(user, locale=locale, slug=slug), None)
+    except Document.DoesNotExist:
+        if locale == settings.WIKI_DEFAULT_LANGUAGE:
+            # Don't repeat the query if it's going to be the same one we just tried.
+            raise Http404
+        # No visible document exists in the requested locale so let's try the default locale.
+        parent = get_visible_document_or_404(
```

Comment:
Basically  we have multiple levels of nesting with functions calling just other functions. Maybe it's better to change the naming of this function altogether and maybe re-approach what each one is doing. Basically this call is doing exactly the same thing with the `try` statement apart the 404. We could simplify this code and all the functions related to visible documents

Reply:
@akatsoulas Good point. I got rid of `get_visible_document_or_fallback_or_404()` and the fallback codes, and refactored `get_visible_document_or_404()` to handle the additional cases. Let's see how you feel about that.